libmamba/src/solv-cpp/pool.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <cassert>
+#include <limits>
+
+#include <solv/pool.h>
+#include <solv/poolid.h>
+#include <solv/pooltypes.h>
+#include <solv/repo.h>
+#include <solv/selection.h>
+
+#include "solv-cpp/pool.hpp"
+
+namespace mamba::solv
+{
+    void ObjPool::PoolDeleter::operator()(::Pool* ptr)
+    {
+        ::pool_free(ptr);
+    }
+
+    ObjPool::ObjPool()
+        : m_pool(::pool_create())
+    {
+    }
+
+    ObjPool::~ObjPool() = default;
+
+    auto ObjPool::raw() -> ::Pool*
+    {
+        return m_pool.get();
+    }
+
+    auto ObjPool::raw() const -> const ::Pool*
+    {
+        return m_pool.get();
+    }
+
+
+    auto ObjPool::disttype() const -> DistType
+    {
+        return raw()->disttype;
+    }
+
+    void ObjPool::set_disttype(DistType dt)
+    {
+        pool_setdisttype(raw(), dt);
+    }
+
+    auto ObjPool::find_string(std::string_view str) const -> std::optional<StringId>
+    {
+        assert(str.size() <= std::numeric_limits<unsigned int>::max());
+        ::Id const id = ::pool_strn2id(
+            const_cast<::Pool*>(raw()),  // Safe because we do not create
+            str.data(),
+            static_cast<unsigned int>(str.size()),
+            /* .create= */ 0
+        );
+        return (id == 0) ? std::nullopt : std::optional(id);
+    }
+
+    auto ObjPool::add_string(std::string_view str) -> StringId
+    {
+        assert(str.size() <= std::numeric_limits<unsigned int>::max());
+        // Note: libsolv cannot report failure to allocate
+        ::Id const id = ::pool_strn2id(
+            raw(),
+            str.data(),
+            static_cast<unsigned int>(str.size()),
+            /* .create= */ 1
```

Comment:
Consider using a named constant to clarify these calls parameters? An `enum class`?

Reply:
This is calling a libsolv function that takes an `int` as a parameter (to represent a bool). I don't see how an enum would apply.

Reply:
It's more for making clear for the reader what is the intend. Here:
```
enum class SomeRelevantName : bool { do_not_create, create };

pool_strn2id( ... , SomeRelevantName.create); // make it clear what is the meaning of the value, instead of 1/0 or true/false.
```